src/SleepRepository.js

```diff
@@ -0,0 +1,81 @@
+const allSleepQualityAvg = [];
+
+class SleepRepo {
+  constructor(sleepData) {
+    this.sleepData = sleepData;
+  }
+
+  getSleepById(id) {
+    return this.sleepData.filter(sleep => sleep.userID === id);
+  }
+
+  calculateAvgSleepStatPerDay(id, stat) {
+    let userSleepData = this.getSleepById(id);
+    let totalStatSum = userSleepData.reduce((sum, sleep) => {
+      sum += sleep[stat];
+      return sum;
+    }, 0);
+    let roundedStatNum = parseInt((totalStatSum / userSleepData.length).toFixed(1));
+    return roundedStatNum;
+  }
+
+  getSleepStatByDate(id, date, stat) {
+    let userSleepData = this.getSleepById(id);
+    let hoursByDate = userSleepData.find(sleep => sleep.date === date);
+    return hoursByDate[stat];
+  }
+
+  getSleepStatsByWeek(id, date, stat) {
+    let userSleepData = this.getSleepById(id);
+    let sleepDates = userSleepData.map(sleep => sleep.date);
+    let indexOfDate = sleepDates.indexOf(date);
+    let sleepByDate = userSleepData.slice(indexOfDate, indexOfDate + 7);
+    return sleepByDate.reduce((obj, sleep) => {
+      obj[sleep.date] = sleep[stat];
+      return obj;
+    }, {});
+  }
+
+  getAvgSleepQualityForUsers() {
+    let avgQuality = this.sleepData.reduce((sum, sleep) => {
+      sum += sleep.sleepQuality;
+      return sum;
+    }, 0);
+    let roundedAvg = Math.round(avgQuality / this.sleepData.length);
+    return roundedAvg;
+  }
+
+  // **DO NOT NEED THESE METHODS AS THEY ARE NOT USED FOR DOM UPDATES**
+
+  // findUserIDs() {
+  //   let result = this.sleepData.reduce((arr, sleep) => {
+  //     if (!arr.includes(sleep.userID)) {
+  //       arr.push(sleep.userID);
+  //     }
+  //     return arr;
+  //   }, [])
+  //   return result;
+  // }
+
+  // getDates(date, id) {
```

Comment:
Great job going above and beyond on this! We will revisit to make this appear on the DOM if time allows